### DISCUSSION TITLE
This app is the most unoptimized writing app I have ever come across.

### DISCUSSION THREAD

**Discussion by Cheburaskminov** [xournalpp/xournalpp#5865] [General]
I have been using Xournal++ for a couple of months now, and frankly speaking this is a very mediocre app in my experience. The app keeps crashing in Ubuntu, which makes writing digital documents an annoying experience. The app can't even recognize the stylus button press, despite my repeated attempts to modify the setting in the Preference tab. I'm using Huion Tablet to write which is well recognized by other apps, which have no problem interpreting the stylus button press. You know how annoying this is, always moving my stylus on the top right corner, pressing the move button to move the zoomed in page of my documents, and then changing back to pen/eraser tool, with the same action. I don't think there is something wrong with my tablet.

The reason why I chose Xournal++ is because OneNote sucks. But it seems like OneNote might be better than Xournal++ when it comes to the ease of using it. My suggestion is that you should optimize your app so that it doesn't crash whenever someone tries to write a document of more than 6 pages and also make your app more compatible with writing tablets.

**Comment by rolandlo** [comment]
Well, it if works so poorly on your device, you're better looking for an alternative. The first one I would try in that case is Rnote. There's no reason to insult the Xournal++ app though as it works fine (without crashes and with working stylus buttons) for a lot of users. If you want to help improving the user experience for Huion Tablet users of Xournal++ first search the issue tracker for "Huion" and if that doesn't already help, open an new issue providing input logs.
